File: rb/features.rb
# ParkingStgallen SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module ParkingStgallenFeatures
  def self.make_feature(name)
    case name
    when "base"
      ParkingStgallenBaseFeature.new
    when "test"
      ParkingStgallenTestFeature.new
    else
      ParkingStgallenBaseFeature.new
    end
  end
end

File: py/features.py
# ParkingStgallen SDK feature factory

from feature.base_feature import ParkingStgallenBaseFeature
from feature.test_feature import ParkingStgallenTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ParkingStgallenBaseFeature(),
        "test": lambda: ParkingStgallenTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
